modules/router-store/src/reducer.ts

```diff
@@ -16,22 +16,20 @@ export type RouterReducerState<
 };
 
 export function routerReducer<
-  T extends BaseRouterStoreState = SerializedRouterStateSnapshot
->(
-  state: RouterReducerState<T> | undefined,
-  action: Action
-): RouterReducerState<T> {
+  RouterState extends BaseRouterStoreState = SerializedRouterStateSnapshot,
+  Result = RouterReducerState<RouterState>
```

Comment:
Should the `Result` extend `RouterReducerState`? Otherwise anything could be pushed into in.
```suggestion
  Result extends RouterReducerState<RouterState> = RouterReducerState<RouterState>
```